src/testFixtures/java/org/opensearch/neuralsearch/BaseNeuralSearchIT.java

```diff
@@ -1057,7 +1057,13 @@ protected void wipeOfTestResources(
             deleteSearchPipeline(searchPipeline);
         }
         if (modelId != null) {
-            deleteModel(modelId);
+            try {
+                deleteModel(modelId);
```

Comment:
Can we make this retryable with max_retry time, e.g. 5, one retry might not helpful.

Reply:
I think the flaky test is caused by some status sync issue in ml-commons. The model is undeployed but it's state is still deployed. And I've create a ticket about this: https://github.com/opensearch-project/ml-commons/issues/1986. For this case, one more call will help the model sync the model state. And I haven't get this exception anymore locally or in github CI after making the changes here.
How about changing this retry to 5 times if we meet this flaky test again? And we should also find the root cause in ml-commons.

Reply:
> We don't need this work as if you see the definition of deleteModel then it already checks for undeployed state 5 times.

From our side we do check the model states multiple times. But from the ml-commons side, the model management system sometimes doesn't sync the model state and need another `undeploy` API call to refresh the model state. You can check the ticket I posted here for reference, the response for getTask API was frozen, no field was updated even includes the `last_update_time`. In this case no matter how many times we call the `getTask` API, we can not get the latest state.